### ISSUE TITLE
Valid YAML file triggers throw parse_error("Failed to convert a scalar to a floating point value", m_line, m_indent);

### ISSUE THREAD

**Issue by tomwpid** [fktn-k/fkYAML#428]
### Description

Trying to load a large but valid YAML file results in the following exception.

throw parse_error("Failed to convert a scalar to a floating point value", m_line, m_indent);

### Reproduction steps

1. Download the problem file [Problem YAML.zip](https://github.com/user-attachments/files/17792281/Problem.YAML.zip)
2. Try to deserialise the file as below.
3. Observe the exception being thrown

### Expected vs. actual results

Expected: YAML should load without error
Actual: YAML cannot be loaded

### Minimal code example

```Shell
// Open a YAML file. Other streams or strings are also usable as an input.
td::ifstream ifs("invNames.yaml");

// Deserialize the loaded file contents.
fkyaml::node RootNode = fkyaml::node::deserialize(ifs);
```

### Error messages

```Shell
throw parse_error("Failed to convert a scalar to a floating point value", m_line, m_indent);
```

### Compiler and operating system

Compiler: MSVC v143 - v14.38 - 17.8 OS: Windows

### Library version

0.3.14

### Validation

- [ ] The bug also occurs if the latest version from the [`develop`](https://github.com/fktn-k/fkYAML/tree/develop) branch is used.
- [ ] I can successfully [compile and run the unit tests](https://github.com/fktn-k/fkYAML#how-to-execute-the-unit-tests).

**Comment by fktn-k** [comment]
@tomwpid   
Thanks for filing the issue!  
The error is reproduced in my local environment too.  
I then investigated the error and found 2 issues in parsing the YAML file.  
It should not take much to fix them, but a workaround for both of them is to single or double quote relevant scalars.  

1. `6E-578` on the line 32882 of invNames.yaml is interpretted as a floating point scalar, but not expressible as the type `double`  
   ```yaml
   # extracted from invNames.yaml (lines 32881-32882)
   - itemID: 30003270
   itemName: 6E-578 # the 6E-578 is interpretted as a floating point scalar but not expressible as the 
   type `double`
   ```

   This causes the error in the description.  
   Although the interpretation itself is correct as per the regular expression for floating point values defined in [the YAML specification](https://yaml.org/spec/1.2.2/#1022-tag-resolution), errors in the string-to-float conversion should be recovered by treating the scalar as a string scalar.  
   However, I'd still recommend the use of either single/double quotes to resolve ambiguity.
2. parsing multiline plain (unquoted) scalars is not yet implemented  
   ```yaml
   # extracted from invNames.yaml (lines 1003361-1003363) as an example
   - itemID: 68002739
     itemName: Factories at Jakanerva II - Moon 5 - Sukuuvestaa Corporation Production
       Plant
   ```

   It's just not yet implemented, and the `deserialize()` function currently returns a wrong result with no error...  
   Single/double quoted multiline scalars are known to be parsed as expected.

**Comment by fktn-k** [comment]
@tomwpid   
My pleasure!  

> Do you have any idea on when these issues might be resolved?  

Since each of them seems to be fixable without much work, I guess they would be fixed in a week.

**Comment by tomwpid** [comment]
@fktn-k - That would be amazing if possible! Looking forward to it... thank you!

**Comment by fktn-k** [comment]
@tomwpid   
The first issue has been fixed by the PR #431, so now the parse_error exception shouldn't been thrown with invNames.yaml though implementation for parsing multiline plain (unquoted) scalars is not yet done (I'm working on it now).

**Comment by fktn-k** [comment]
@tomwpid   
The second issue in parsing multiline plain scalars has also been fixed by the PR #432.  
I roughly checked the parse result of the invNames.yaml file with the develop HEAD locally,  
and it seems correct as the following checks now pass successfully:

* The number of sequence items is the same as that of occurrences of "itemID" 
  (since each sequence item has it as a mapping key)
* Each sequence item is a mapping node which has exactly the following 2 kinds of key-value pairs  
  (values are not checked)
  * "itemID" key to an integer scalar node
  * "itemName" key to a string scalar node

Let me know if you find anything unresolved.

**Comment by fktn-k** [comment]
@tomwpid   
I'll close this issue as resolved.  
Feel free to reopen it if you find anything related but still unresolved.
